cyrw/ioctl.go

```diff
@@ -316,43 +316,70 @@ func (d *Device) f2PacketAvail() (bool, uint16) {
 	return false, 0
 }
 
+// Ioctl polling errors.
+var (
+	errNoF2Avail            = errors.New("no packet available")
+	errWaitForCreditTimeout = errors.New("waitForCredit timeout")
+)
+
 // waitForCredit waits for a credit to use for the next transaction
 func (d *Device) waitForCredit(buf []uint32) error {
 	d.trace("waitForCredit")
 	if d.has_credit() {
 		return nil
 	}
-	_, err := d.pollForIoctl(buf) // credit is updated with poll.
-	return err
+	for retries := 0; retries < 10; retries++ {
+		_, _, err := d.tryPoll(buf)
+		// TODO(soypat): ether type error?
```

Comment:
How about instead of awkwardly handling ErrInvalidEtherType here, don't make it an error at all?  In other words, don't return an error in DecodeEventPacket() for bad eth type pkts, just post a debug warning or something, but otherwise skip over that pkt.

In terms of waiting for credits, even the bad eth type pkt counts for returning credits, so we make progress and can queue the next transaction.

For pollForIoctl can just skip over the bad eth type pkt also.

Otherwise all of these changes with tryPoll() look great, thank you.

Reply:
Ya, I don't know...maybe not returning an err isn't right either.  Makes DecodeEventPacket() awkward.

BTW, I'm not seeing a panic because of these bad eth type pkts how is that happening for you?

I see two of them sometime after clm load, but ignored 'em.

Reply:
> how is that happening for you?

yeah same here. depending on how the program runs the panics happen or they don't :shrug:.

> maybe not returning an err isn't right either.

I did consider it... but yeah, seemed very awkward to discard the error at any other level other than the one we're doing it now. Otherwise bubbling up control flow is very awkward, best to just do it with errors. In any case it's good documentation of a problem that we might need to take care of someday

> In terms of waiting for credits, even the bad eth type pkt counts for returning credits 

good point. Will add that logic

Reply:
> > In terms of waiting for credits, even the bad eth type pkt counts for returning credits 
>
> good point. Will add that logic

Actually- my understanding is it is already doing that. It ignores the first if statement and goes to the else if where it finds out it has credits.